step.yml

```diff
@@ -24,20 +24,77 @@ toolkit:
   go:
     package_name: github.com/bitrise-steplib/bitrise-step-xcode-start-simulator
 inputs:
-  - destination: platform=iOS Simulator,name=iPhone 8 Plus,OS=latest
-    opts:
-      title: Device destination specifier
-      summary: Destination specifier describes the simulator device to be started.
-      description: |-
-        Destination specifier describes the simulator device to be started.
+- destination: platform=iOS Simulator,name=iPhone 8 Plus,OS=latest
+  opts:
+    title: Device destination specifier
+    summary: Destination specifier describes the simulator device to be started.
+    description: |-
+      Destination specifier describes the simulator device to be started.
 
-        The input value uses the same format as xcodebuild's `-destination` option.
-      is_required: true
+      The input value uses the same format as xcodebuild's `-destination` option.
+    is_required: true
+
+- erase: "no"
+  opts:
+    title: Erase Simulator before use
+    summary: If enabled, will erase a simulator's contents and settings.
+    description: |-
+      If enabled, will erase a simulator's contents and settings.
+    value_options:
+    - "yes"
+    - "no"
+    is_required: true
+
+- wait_for_boot: "no"
+  opts:
+    title: Wait for simulator to boot
+    summary: If enabled, will wait simulator boot to complete.
+    description: |-
+      If enabled, will wait simulator boot to complete.
+
+      Setting to `yes` makes possible to detect hangs or timeouts when booting simulator.
+      If a timeout occurs, the `BITRISE_IS_SIMULATOR_TIMEOUT` output will be set to true.
+
+      Using `no` (the default) enables to boot the Simulator parallel to other Steps.
+    value_options:
+    - "yes"
+    - "no"
+    is_required: true
+
+# Debugging
+
+- verbose_log: "no"
+  opts:
+    category: Debugging
+    title: Enable verbose logging
+    summary: If this input is set, the Step will print additional logs for debugging.
+    value_options:
+    - "yes"
+    - "no"
+    is_required: true
+
+- wait_for_boot_timeout: 90
+  opts:
+    category: Debugging
+    title: Simulator boot timeout
+    summary: Maximum allowed time for simulator boot (in seconds)
+    description: |-
+      Maximum allowed time for simulator boot (in seconds)
+
+      "Wait for simulator to boot" (`wait_for_boot`) must be set to `yes`.
```

Comment:
Curious about this requirement. 
Why do I have to set two flags to make this timeout happen? 

Would it be cleaner to simply use the presence of a timeout as a signal to the system that we want to wait for boot?

Reply:
Yes it is better to avoid dependencies between inputs. Removed this one. If the timeout is set to 0 then no wait is done.